client/src/ui/molecules/search/index.tsx

```diff
@@ -28,21 +42,71 @@ export function Search({
 }) {
   const [value, setValue] = useQueryParamState();
   const [isFocused, setIsFocused] = useState(false);
-  const [defaultSelection, setDefaultSelection] = useState([0, 0] as const);
 
-  const searchProps = useMemo(
+  const gleanClick = useGleanClick();
+  const lastFocus = useRef(isFocused);
+  const lastValue = useRef(value);
+  const [hasOpened, setHasOpened] = useState(false);
+  const [hasEdited, setHasEdited] = useState(false);
+  const [hasClicked, setHasClicked] = useState(false);
+
+  const name = isHomepageSearch ? "hp" : "top";
+
+  const measure = useCallback(
+    (action: string, value?: string) => {
+      gleanClick(
+        `${SEARCH}: ${name} ${action}` + (value ? ` -> ${value}` : "")
+      );
+    },
+    [gleanClick, name]
+  );
+
+  useEffect(() => {
+    if (isFocused !== lastFocus.current) {
+      if (isFocused) {
+        measure(SEARCH_ACTIONS.OPEN);
+        setHasOpened(true);
+      } else {
+        if (!hasClicked) {
+          measure(SEARCH_ACTIONS.CLOSE);
+        }
+        // Search completed.
+        setHasOpened(false);
+        setHasEdited(false);
+        setHasClicked(false);
+      }
+      lastFocus.current = isFocused;
+    }
+  }, [isFocused, hasOpened, hasClicked, measure]);
+
+  useEffect(() => {
+    if (value !== lastValue.current) {
+      if (!hasEdited) {
+        measure(SEARCH_ACTIONS.EDIT);
+        setHasEdited(true);
+      }
+      lastValue.current = value;
+    }
+  }, [value, hasEdited, measure]);
+
+  const searchProps: SearchNavigateWidgetProps = useMemo(
     () => ({
       id,
       inputValue: value,
       onChangeInputValue: (value) => setValue(value),
       isFocused,
-      onChangeIsFocused: (isFocused) => {
-        setIsFocused(isFocused);
+      onChangeIsFocused: (isFocused) => setIsFocused(isFocused),
+      defaultSelection: [0, 0],
+      onResultClick: (
+        value: string,
+        event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
+      ) => {
+        gleanClick(`${QUICKSEARCH}: ${value}`);
+        measure(SEARCH_ACTIONS.CLICK, event.currentTarget.href);
```

Comment:
Unfortunately, the `href`s in the search are fully-qualified URLs, which takes unnecessary space in the value.

Two options:
1. Parse the URL to get just the path.
2. Use relative URLs for the search result items.

Reply:
The search result items do use relative URLs, but `href` returns the parsed/resolved value.

Fixed in a745c238b7.